jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/HTTP2Stream.java

```diff
@@ -537,23 +537,23 @@ public Data readData()
     @Override
     public void demand()
     {
-        boolean process = false;
+        boolean dispatch = false;
         try (AutoLock ignored = lock.lock())
         {
             dataDemand = true;
             if (dataStalled && !dataQueue.isEmpty())
             {
                 dataStalled = false;
-                process = true;
+                dispatch = true;
             }
         }
         if (LOG.isDebugEnabled())
-            LOG.debug("Demand, {} data processing for {}", process ? "proceeding" : "stalling", this);
-        if (process)
-            processData();
+            LOG.debug("Demand, {} data processing for {}", dispatch ? "proceeding" : "stalling", this);
+        if (dispatch)
+            processData(true);
```

Comment:
I'd like a comment explaining why we must dispatch here, similar to the one in `HttpStreamOverHTTP2`.